<?xml version="1.0"?>
<robot  name="rbs_arm" >
 <link  name="world" />
 <joint  name="world_rbs_arm_joint"  type="fixed" >
  <origin  rpy="0 0 0"  xyz="0 0 0" />
  <parent  link="world" />
  <child  link="rbs_arm_link_0" />
 </joint>
 <link  name="rbs_arm_link_0" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00503302470272442"  ixy="0.000343817346410954"  izz="0.00405858207282473"  ixz="-4.74990755448368e-06"  iyz="-2.3099255620051e-05"  iyy="0.00337962410057753" />
   <origin  xyz="-0.000297002857922682 0.0964721185617698 -0.000361033370053684" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/start_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/start_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_1"  type="revolute" >
  <parent  link="rbs_arm_link_0" />
  <child  link="rbs_arm_link_1" />
  <origin  rpy="0.0 0.0 0.0"  xyz="0.0 0.0 0.17833" />
  <axis  xyz="0 0 1" />
  <limit  upper="3.14159"  effort="78"  lower="-3.14159"  velocity="0.52" />
 </joint>
 <link  name="rbs_arm_link_1" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00147695259043549"  ixy="-2.66894744420299e-05"  izz="0.00087582892706912"  ixz="-4.40871314563273e-05"  iyz="-3.19001462979333e-05"  iyy="0.00135500487881796" />
   <origin  rpy="0.0 0.0 0.0"  xyz="-9.7531539777207e-06 -0.000888494418875867 0.0342332199538358" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/fork_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/fork_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_2"  type="revolute" >
  <parent  link="rbs_arm_link_1" />
  <child  link="rbs_arm_link_2" />
  <origin  rpy="0.0 0.0 0.0"  xyz="0.1 0.0 0.094" />
  <axis  xyz="0 1 0" />
  <limit  upper="3.14159"  effort="78"  lower="-1.5708"  velocity="0.52" />
 </joint>
 <link  name="rbs_arm_link_2" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00147695259043549"  ixy="-2.66894744420299e-05"  izz="0.00087582892706912"  ixz="-4.40871314563273e-05"  iyz="-3.19001462979333e-05"  iyy="0.00135500487881796" />
   <origin  rpy="0.0 0.0 0.0"  xyz="-9.7531539777207e-06 -0.000888494418875867 0.0342332199538358" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/main_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/main_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_3"  type="revolute" >
  <parent  link="rbs_arm_link_2" />
  <child  link="rbs_arm_link_3" />
  <origin  rpy="0.0 0.0 0.0"  xyz="0.0 0.0 0.133" />
  <axis  xyz="0 0 1" />
  <limit  upper="3.14159"  effort="78"  lower="-3.14159"  velocity="0.52" />
 </joint>
 <link  name="rbs_arm_link_3" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00147695259043549"  ixy="-2.66894744420299e-05"  izz="0.00087582892706912"  ixz="-4.40871314563273e-05"  iyz="-3.19001462979333e-05"  iyy="0.00135500487881796" />
   <origin  rpy="0.0 0.0 0.0"  xyz="-9.7531539777207e-06 -0.000888494418875867 0.0342332199538358" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/fork_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/fork_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_4"  type="revolute" >
  <parent  link="rbs_arm_link_3" />
  <child  link="rbs_arm_link_4" />
  <origin  rpy="0.0 0.0 0.0"  xyz="0.1 0.0 0.094" />
  <axis  xyz="0 1 0" />
  <limit  upper="3.14159"  effort="78"  lower="-1.5708"  velocity="0.52" />
 </joint>
 <link  name="rbs_arm_link_4" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00147695259043549"  ixy="-2.66894744420299e-05"  izz="0.00087582892706912"  ixz="-4.40871314563273e-05"  iyz="-3.19001462979333e-05"  iyy="0.00135500487881796" />
   <origin  rpy="0.0 0.0 0.0"  xyz="-9.7531539777207e-06 -0.000888494418875867 0.0342332199538358" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/main_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/main_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_5"  type="revolute" >
  <parent  link="rbs_arm_link_4" />
  <child  link="rbs_arm_link_5" />
  <origin  rpy="0.0 0.0 0.0"  xyz="0.0 0.0 0.133" />
  <axis  xyz="0 0 1" />
  <limit  upper="3.14159"  effort="78"  lower="-3.14159"  velocity="0.52" />
 </joint>
 <link  name="rbs_arm_link_5" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00147695259043549"  ixy="-2.66894744420299e-05"  izz="0.00087582892706912"  ixz="-4.40871314563273e-05"  iyz="-3.19001462979333e-05"  iyy="0.00135500487881796" />
   <origin  rpy="0.0 0.0 0.0"  xyz="-9.7531539777207e-06 -0.000888494418875867 0.0342332199538358" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/fork_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/fork_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_6"  type="revolute" >
  <parent  link="rbs_arm_link_5" />
  <child  link="rbs_arm_link_6" />
  <origin  rpy="0.0 0.0 0.0"  xyz="0.1 0.0 0.094" />
  <axis  xyz="0 1 0" />
  <limit  upper="3.14159"  effort="78"  lower="-1.5708"  velocity="0.52" />
 </joint>
 <link  name="rbs_arm_link_6" >
  <inertial >
   <mass  value="1.88031044620482" />
   <inertia  ixx="0.00147695259043549"  ixy="-2.66894744420299e-05"  izz="0.00087582892706912"  ixz="-4.40871314563273e-05"  iyz="-3.19001462979333e-05"  iyy="0.00135500487881796" />
   <origin  rpy="0.0 0.0 0.0"  xyz="-9.7531539777207e-06 -0.000888494418875867 0.0342332199538358" />
  </inertial>
  <visual >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/visual/ee_link.dae" />
   </geometry>
  </visual>
  <collision >
   <geometry >
    <mesh  filename="/home/bill-finger/rbs_ws/install/rbs_arm/share/rbs_arm/meshes/collision/ee_link.stl" />
   </geometry>
  </collision>
 </link>
 <joint  name="rbs_arm_joint_tool0"  type="fixed" >
  <origin  rpy="0 0 0"  xyz="0 0 0.11" />
  <parent  link="rbs_arm_link_6" />
  <child  link="rbs_arm_tool0" />
 </joint>
 <link  name="rbs_arm_link_tool0" />
 <ros2_control  name="rbs_arm_joint_1"  type="actuator" >
  <hardware >
   ign_ros2_control/IgnitionSystem
  </hardware>
  <joint  name="rbs_arm_joint_1" >
   <command_interface  name="position" />
   <command_interface  name="velocity" />
   <state_interface  name="position" />
   <state_interface  name="velocity" />
   <state_interface  name="effort" />
  </joint>
 </ros2_control>
 <ros2_control  name="rbs_arm_joint_2"  type="actuator" >
  <hardware >
   ign_ros2_control/IgnitionSystem
  </hardware>
  <joint  name="rbs_arm_joint_2" >
   <command_interface  name="position" />
   <command_interface  name="velocity" />
   <state_interface  name="position" />
   <state_interface  name="velocity" />
   <state_interface  name="effort" />
  </joint>
 </ros2_control>
 <ros2_control  name="rbs_arm_joint_3"  type="actuator" >
  <hardware >
   ign_ros2_control/IgnitionSystem
  </hardware>
  <joint  name="rbs_arm_joint_3" >
   <command_interface  name="position" />
   <command_interface  name="velocity" />
   <state_interface  name="position" />
   <state_interface  name="velocity" />
   <state_interface  name="effort" />
  </joint>
 </ros2_control>
 <ros2_control  name="rbs_arm_joint_4"  type="actuator" >
  <hardware >
   ign_ros2_control/IgnitionSystem
  </hardware>
  <joint  name="rbs_arm_joint_4" >
   <command_interface  name="position" />
   <command_interface  name="velocity" />
   <state_interface  name="position" />
   <state_interface  name="velocity" />
   <state_interface  name="effort" />
  </joint>
 </ros2_control>
 <ros2_control  name="rbs_arm_joint_5"  type="actuator" >
  <hardware >
   ign_ros2_control/IgnitionSystem
  </hardware>
  <joint  name="rbs_arm_joint_5" >
   <command_interface  name="position" />
   <command_interface  name="velocity" />
   <state_interface  name="position" />
   <state_interface  name="velocity" />
   <state_interface  name="effort" />
  </joint>
 </ros2_control>
 <ros2_control  name="rbs_arm_joint_6"  type="actuator" >
  <hardware >
   ign_ros2_control/IgnitionSystem
  </hardware>
  <joint  name="rbs_arm_joint_6" >
   <command_interface  name="position" />
   <command_interface  name="velocity" />
   <state_interface  name="position" />
   <state_interface  name="velocity" />
   <state_interface  name="effort" />
  </joint>
 </ros2_control>
</robot>
